### PR TITLE
Removes sleeping carp's hidden damage resistance + 40 brute crit punches, lowers price

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Hi all, usually I stay far away from balance memes and all that, but during my work on wounds I was made aware that sleeping carp was completely broken with wounds, since it dealt so much damage. I went in to see if I needed to make adjustments for balance, and was shocked at what I found. Apparently sleeping carp has had lots of changes since I last saw it.

For the uninitiated, sleeping carp grants:

- 60% global brute resist
- 30% global burn resist (which the code tries to play off as a "lesser" strength)
- 50% global stamina resist (I should point out the author to these changes is adamantly trying to change stunbatons to do solely stamina damage)
- 70% global stun resist
- 70% global pressure resist
- 70% global cold resist
- Immunity to dismemberment
- Immunity to all embedding, injections, and caltrops
- Ability to deflect all projectiles while throw mode is enabled

with no visible tell. The only thing that sleeping carp users are actually hurt more by is being on fire, though that's reduced of course by the burn mod (x2 * x0.3 = 1.4x damage from fire), and maybe pressure mod too. I'd also like to point out that all of these incredible resistances are before armor is even applied, so you can easily push those all up by like 20-30 percentage points if they get their hands on armor.

I'm not really interested in arguments about whether or not there's a specific set of counters that are actually effective to this, because a few people have offered very long lists of very specific ways that sleeping carp can be defeated, I'm removing the resistances because it's bad gameplay and the ultimate in munchkinry. All of those buffs above come before armor is applied, before stims and meds are applied, and again are totally invisible to the vast majority of players who don't specifically follow the meta or randomly code dive to learn that there are only certain very specific counters to it and if you don't have that plan already worked out then you're just screwed.

This also removes the 10% chance to do 20 extra brute damage for a total of 40 with every second punch because what the hell is that

## Why It's Good For The Game
At least back when sleeping carp was first added and your only defensive buff was deflecting bullets you could recognize that pretty quickly and move in with a baton instead. Even after it was explained to me like thrice how you're supposed to counter a carp user I still couldn't tell you what you're supposed to do, so there's no point in subjecting normal players to this mess.

## Changelog
:cl: Ryll/Shaps
del: Sleeping carp no longer grants outright damage resistances to all forms of damage. Its price has been decreased to 12TC
/:cl:

### CODE REVIEW

**Comment by doguinh0** [comment]
no no no no no no no no no

**Comment by Ryll-Ryll** [comment]
Make it something that doesn't have invisible 30-70% resists on everything then

**Comment by wesoda25** [comment]
So basically this is the trigger to a pr balance war. Its too radical but will get people scared enough to accept a PR which gives us a weaker form of original carp (lemme guess you can’t aggressive grab people?). Nice job

**Comment by necromanceranne** [comment]
Rest in Spaghet fishmen.

**Comment by doguinh0** [comment]
> Make it something that doesn't have invisible 30-70% resists on everything then

that's better than removing it, but still, it's not that used anyways

**Comment by optimumtact** [comment]
It should be changed to just deflect projectiles again, all these resistances are stupid

**Comment by MoonFalcon** [comment]
Carp without resistances and forced deflect is going to be a really bad weapon. People will be able to "taser test" people to see if they have carp (like they used to all the time) and the resistances are the only thing that make carp viable. 

The attacks themselves are just not good enough to justify 17 tc. You will now be able to be killed by two spear throws as soon as you are outed and there is nothing you can do about it.

**Comment by MoonFalcon** [comment]
not to mention that using throw to deflect was really the only skill based / interesting thing carp had going for it.

**Comment by optimumtact** [comment]
That's okay you don't have to use it

**Comment by necromanceranne** [comment]
I tried to do something different and see how it went. If it isn't any good then quite honestly, scarp probably should be killed off like in this pr. Straight up competing in a baton joust doesn't work when the martial art has no way of bring you down in a timely manner if it can't soak hits, so you just lose the moment anyone moves at you with one because you have no options and you can stop someone screaming for help because you lack a silence. The combo system is kind of not great when another opponent only requires touching you once to your two or three or whatever, and even when I changed it to two moves it still isn't amazing.

I considered alternatives like adding chokeholds to keep people quiet as you snapped their or something, so it was strictly an assassination tool and not something you can use in a straight fight, but then I realized that's just CQC and I don't want to be responsible for adding CQC for traitors. So I went harder in the other direction.

I tried my best to get a good sense of whether it was good or not. Most people said it was harder and many said it was only good for environmental shenanigans. It seemed like I made it a reasonable niche choice.

**Comment by Ryll-Ryll** [comment]
I'll modify it to drop the resistances

**Comment by necromanceranne** [comment]
I would like to say that given I have a stun baton pr up, if that goes even somewhat through then my concerns are probably not needed and this **might** be fine though you may want to remove nopiercing and nodismember since those were mostly to theme.

**Comment by ATH1909** [comment]
I'd like to point out that, as far as I'm aware, the sleeping carp resistances stack multiplicatively with armor, not additively.

Yeet the random crits (which remind me an awful lot of TF2's random crits), because combat RNG=bad, but keep the damage resistances and price.

File: code/datums/martial/sleeping_carp.dm [hunk 1]
```diff
@@ -23,26 +23,18 @@
...
-		crit_damage += 20
-		playsound(get_turf(D), 'sound/weapons/bite.ogg', 50, TRUE, -1)
-		D.visible_message("<span class='warning'>[D] is sent reeling as the blow strikes them with inhuman force!</span>", "<span class='userdanger'>You are struck with incredible precision by [A]!</span>")
-		log_combat(A, D, "critcal strong punched (Sleeping Carp)")//log it here because a critical can swing for 40 force and it's important for the sake of how hard they hit
-	else
-		playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
-		log_combat(A, D, "strong punched (Sleeping Carp)")//so as to not double up on logging
-	D.apply_damage(20 + crit_damage, A.dna.species.attack_type, affecting)
+	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
+	log_combat(A, D, "strong punched (Sleeping Carp)")//so as to not double up on logging
```
Comment (ATH1909) [inline code/datums/martial/sleeping_carp.dm:36]: remove the comment here

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -584,10 +584,10 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 
 /datum/uplink_item/stealthy_weapons/martialarts
 	name = "Martial Arts Scroll"
-	desc = "This scroll contains the secrets of an ancient martial arts technique. You will master unarmed combat, \
-			gain skin as hard as steel and swat bullets from the air, but you also refuse to use dishonorable ranged weaponry."
+	desc = "This scroll contains the secrets of an ancient martial arts technique. You will master unarmed combat \
+			and the ability swat bullets from the air, but you also refuse to use dishonorable ranged weaponry."
```
Comment (ATH1909) [inline code/modules/uplink/uplink_items.dm:588]:
```suggestion
			and gain the ability to swat bullets from the air, but you will also refuse to use dishonorable ranged weaponry."
```

**Comment by ATH1909** [comment]
> So basically this is the trigger to a pr balance war. Its too radical but will get people scared enough to accept a PR which gives us a weaker form of original carp (lemme guess you can’t aggressive grab people?). Nice job

Hasn't sleeping carp already lost its instant aggressive grab?

**Comment by necromanceranne** [comment]
It did!

**Comment by ATH1909** [comment]
Also, with this PR, buying sleeping carp+gloves of the north star now only costs 20 TC, so you wouldn't need a discount (or another traitor's TC) in order to use that combo, which is a bit of a yikes.

**Comment by gitbirb** [comment]
> Also, if this PR gets merged in its current state, buying sleeping carp+gloves of the north star will only costs 20 TC, so you won't need a discount (or another traitor's TC) in order to use that combo, which is a bit of a yikes.

A pr got merged a while ago that made north stars into a martial art, so it doesnt combo anymore with sleeping carp. All sleeping carp does currently would let northstar guy deflect bullets while  throw mode is on and also carp resistances, making it significantly nerfed from its original inception, but not overpowered. So it really doesn't matter anymore if the resistances are getting fucked up, the only thing it would add to someone with northstar gloves is deflecting projectiles and that's it, which you could do a better form of this combo entirely with noslips and a DEsword anyways.

**Comment by necromanceranne** [comment]
That northstar pr wasn't merged.

**Comment by ExcessiveUseOfCobblestone** [comment]
> rng in combat is bad

yeah cause then all that powergaming may be for naught!

**Comment by uomo91** [comment]
>People will be able to "taser test" people to see if they have carp (like they used to all the time)

Why not just tie deflect to an action button with some cooldown for reactivation?
Makes stealth viable, but not with the ability to instantly start using it again right after deactivating.

**Comment by Ryll-Ryll** [comment]
I don't mind RNG in combat, and have worked it into a few of my own features, but outright putting in a 10% chance to do +20 damage is the worst kind of RNG

**Comment by evilcyborgmrt** [comment]
what's munchkinry?

**Comment by HomocidlePancakes** [comment]
1)Newfriend above
2)So if this is getting gutted, why not just bring back old carp?

**Comment by Tlaltecuhtli** [comment]
old carp just needed some twearking (remove insta grab + insta grab into a sleep) and some cool new moves to  be the dance dance revolution two of syndicate items

**Comment by HomocidlePancakes** [comment]
Tlal, you are mistaking carp for CQC. Old carp didn’t have a sleep, just a stun.

**Comment by Shadowflame909** [comment]
>no no no no no no no no no, also, the syndicate is supposed to be stronger than the station (as a whole, not if everyone was totally cooperative)

Maybe early game.

But you should suck eggs if you think that simply paying 16 tc at roundstart for your OP wombo combo should do anything against a mech or a sec officers meteor slug shotgun.

Telecrystals rapidly diminish in value the longer a round goes on.

**Comment by MarioWizard119** [comment]
I always loved using carp scroll as a gimmick weapon as a traitor or nukeop because of its main property, the ability to deflect projectiles, since it forces people into melee. I'm not opposed to the addition of needing throw active to deflect since it makes carp concealable and keeps you from cheesing the melee fight by using weapons such as the syndie toolbox or flashes. What really made old carp cancerous was the 100% armor piercing on your punches combined with the instant aggressive grabs which made 1v1ing someone with carp suicide, even if you took precauitions to fight someone in melee.

So what does new carp have in exchange? Carp now is a near unkillable tank which can only be defeated if you can somehow get the AI to plasmaflood the user.

This is one of the few nerfs I can actually stand by. The resistances undermine the gimmick and make carp scroll stupid strong.

**Comment by ATH1909** [comment]
> I'm not opposed to the addition of needing throw active to deflect since it makes carp concealable and keeps you from cheesing the melee fight by using weapons such as the syndie toolbox or flashes. 

Er, that's already how it currently works.

**Comment by ATH1909** [comment]
> > People will be able to "taser test" people to see if they have carp (like they used to all the time)
> 
> Why not just tie deflect to an action button with some cooldown for reactivation?
> Makes stealth viable, but not with the ability to instantly start using it again right after deactivating.

It's best to make carp users choose between deflecting OR attacking, so that it'll still be possible to hit a carp user with ranged weapons if you can bait (heh heh, "bait") them into rushing at you (or someone else) in melee. It adds a layer of strategy to both playing as and fighting a sleeping carp user.

**Comment by OverlyGeneric** [comment]
Wasn't the point of old carp deflect to force the opponent to come closer?

**Comment by Tharcoonvagh** [comment]
khm carpgloves

**Comment by Shadowflame909** [comment]
Why not just revert it to old carp?

**Comment by MarioWizard119** [comment]
Because 1 click win aggressive grabs and 100 percent armor piercing on punch, which has been practically turned into an undroppable circsaw, makes 1v1ing someone with old carp suicide.

**Comment by necromanceranne** [comment]
Current scarp is still sorta suicide to 1v1 but it is theoritically doable just very fucking hard.

It's just less rocket taggy.

**Comment by HomocidlePancakes** [comment]
> Because 1 click win aggressive grabs and 100 percent armor piercing on punch, which has been practically turned into an undroppable circsaw, makes 1v1ing someone with old carp suicide.

Words of an unrobust dude.

All it takes to defeat a carp user is a bola. They are not hard to get and you go in to any combat situation unprepared, it’s entirely your fault if you die.

**Comment by wesoda25** [comment]
facing someone with old carp was 10x easier than new carp

**Comment by Ryll-Ryll** [comment]
A wielded spear, the best weapon a vast majority of the crew can get, does 7.2 damage before armor to a sleeping carp user. A wielded fireaxe, the single most powerful crew weapon on station, does 9.8 damage. This is, again, before armor, which can easily knock those down another 30-40% from there. Please stop being disingenuous and saying the average player stands any sort of chance against those kinds of odds.

**Comment by ATH1909** [comment]
> A wielded spear, the best weapon a vast majority of the crew can get, does 7.2 damage before armor to a sleeping carp user. A wielded fireaxe, the single most powerful crew weapon on station, does 9.6 damage. This is, again, before armor, which can easily knock those down another 30-40% from there. With a sec/miner jumpsuit and a HoS coat, that 24 damage is now 5.7. Please stop being disingenuous and saying the average player stands any sort of chance against those kinds of odds.

What about lit welders?

**Comment by Shadowflame909** [comment]
This makes me miss throwing bomb spears.

Also... @necromanceranne That 20 tc Immunity to one damage type hardsuit doesn't seem so silly now!

**Comment by ATH1909** [comment]
Alright, you convinced me about the brute and burn damage resistances, but what about the stun resistance, the stamina resistance, and the TRAIT_STUNRESISTANCE trait? Can we keep those?

**Comment by necromanceranne** [comment]
@Shadowflame909 only if it's nodrop and makes you take double damage to two other damage types.

@ATH1909 A lit welder is reduced to 10.5. It is also subject to armor but 10 damage is a pretty typical amount. If we take this against someone in a sec vest, it's going to be reduced to about 6 damage. Miners are an extreme situation since they get up close to the damage caps and even withou scarp are an incredibly overtanky nuisance.

One thing I was going to do in the scarp change alongside batons was drastically increase the stamina damage scarp takes. At a 20% reduction it's more or less like walking around with a cham vest on. But 30 damage batons would end up doing 27. Not accounting armor. And disablers would be good too, as would rubbershot.

The reason it is 50% is because batons do 60 stamina damage and ignore armor. Given they are the predominate melee weapon, they get reduced to 30 damage against a sleeping carp user right now, ignoring all armor. A baton rush can do it. And hell, even one guy can do it. And stunprods are available to anyone who can make them. When I said 'stun batons are the one thing that can fight them', it's because prods and batons ignore armor cheese.

**Comment by ExcessiveUseOfCobblestone** [comment]
why does kung fu stop dismemberment, so stupid.

People really can't help but convert inches to miles, then act surprise when we want to throw the entire ruler out.

**Comment by wesoda25** [comment]
This is a balance PR not removal

**Comment by necromanceranne** [comment]
Thinking about this, it really does make me worried about a bunch of scarpies running around with fist of the north star every round. Even if it isn't going to crit, 15-20 damage rapidly applied and completely armor piercing is going to get fucking obnoxious. Sheer glass cannon scarp is definitely a lot closer to the original but making gloves of the north star a no-brainer purchase is not really any better to what it was before.

**Comment by wesoda25** [comment]
Didn’t we make north star a martial art so that people can’t combine it with other stuff?

**Comment by necromanceranne** [comment]
It got closed.

**Pull request merged**